# Quiz Capítulo 5.

* **Estudiantes**

## Pregunta 1.

¿Cuál de las siguientes afirmaciones sobre el aprendizaje supervisado es incorrecta?

1. El alumno supervisado debe hacer predicciones correctamente con los datos proporcionados a partir de los datos de entrenamiento.

2. La precisión es la relación entre lo que el modelo clasifica como verdadero y lo que realmente clasifica como verdadero. Esto se puede llamar tasa de respuesta.

3. El índice de recuerdo es la relación entre lo que el modelo predice que es verdadero y lo que realmente es cierto.

4. El índice de recuperación se calcula como ${\frac{t p}{t p+f p}}\ {\mathrm{es~la~expresión~para~precisión}}$

<Font color=red>[Respuesta]</Font>

La afirmación incorrecta es 4

ya que le expresión para la recuperacion se calcula TP / TP+FN


## Pregunta 2.

¿Cuál es el método para estimar los coeficientes de regresión en el análisis de regresión?

<Font color=red>[Respuesta]</Font>

El método para estimar coeficientes de regreseiónn en el análisis de regresión es el método de minimos cuadrados ordinarios el MCO, ya que minimiza la suma de los cuadrados entre el valor observado y el valor predicho por el modelo

## Pregunta 3.

¿Cuál de las siguientes afirmaciones sobre el sobreajuste es incorrecta?

1. Una característica de un patrón que se encuentra en un conjunto de datos específico y que existe solo en ese conjunto de datos, o una característica de un patrón que no se generaliza y existe solo en un dato específico se denomina sobreajuste.

2. Los métodos de extracción utilizados para resolver el problema de sobreajuste incluyen el método de reserva, el método de validación cruzada, el método de arranque, etc.

3. En el método de reserva, la validación cruzada se realiza dividiendo los datos de entrenamiento para el aprendizaje y la construcción del modelo y los datos de verificación para la evaluación del desempeño, y los resultados de los datos de verificación se usan solo para la medición del desempeño sin afectar el modelo.

4. El método de validación cruzada es un método para volver a seleccionar datos de entrenamiento repetidamente y basado en la extracción de restauración, y es adecuado cuando la cantidad total de datos no es grande.

<Font color=red>[Respuesta]</Font>


La afirmación incorrecta es la 3.

el método de reserva se basa en se dividir el conjunto de datos en un conjunto de entrenamiento y un conjunto de prueba, el conjunto de entrenamiento se usa para entrenar el modelo y el de prueba se usa para evaluar el modelo

## Pregunta 4.

¿Cuál de las siguientes afirmaciones sobre el modelo de árbol de decisión es incorrecta?

1. Los árboles de decisión se utilizan para la predicción de quiebras corporativas, la predicción del tope del precio de las acciones, la predicción del tipo de cambio y la predicción de las perspectivas económicas.

2. El algoritmo CHAID (detección automática de interacción de chi-cuadrado) es un algoritmo que realiza la separación mediante chi-cuadrado o prueba F.

3. El algoritmo CART (árboles de clasificación y regresión) es un algoritmo que realiza la separación utilizando el índice de Gini, y 100 representa un nodo perfectamente puro entre 0 y 100.

4. El índice de entropía del algoritmo C4.5 se puede obtener utilizando la estadística de prueba de razón de verosimilitud en la distribución polinómica y restando la entropía del nodo hijo de la entropía del nodo padre.

<Font color=red>[Respuesta]</Font>

La afirmación incorrecta es el 2.

El algoritmo CHAID es la construcción de árboles de decisión que se basa en pruebas de independencia de chi-cuadrado, por lo que no se uisa la prueba F, así que es incorrecta ya que la afirmación menciona la prueba F

## Pregunta 5.

Cuando entrené un clasificador SVM usando el kernel RBF, parece no adaptarse al conjunto de entrenamiento.<br>
¿Debo aumentar o disminuir gamma? <br> ¿Qué pasa con C?

<Font color=red>[Respuesta]</Font>

GAMMA:

Si se aumenta el gamma, el modelo seajusta a los datos de entrenamiento lo que haria un sobreajuste si el valor de gamma es demasiado alto, pero si se disminuye el gamma, el modelo será menos sensible a los datos de entrenamiento.

C:

Si  se aumenta el C, el modelo penaliza más los errores de clasificación, lo que haría al modelo más ajustado a los datos de entrenamiento.
Pero si se disminuye C, el modelo se enfoca más en maximizar el margen, lo que hace mejorar la generalización

## Pregunta 6.

Utilice el algoritmo KNN para clasificar los datos en el código siguiente y calcular la exactitud, la llamada y la precisión. Establezca el valor de adyacencia en 5.

In [ ]:
import numpy as np
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.neighbors import KNeighborsClassifier
from sklearn import metrics, preprocessing

# Load data (descargar los datos)
data = load_breast_cancer()
X = data['data']
Y = 1 - data['target']  # 0 = 'benign' and 1 = malignant.  # 0 = benigno y 1 = maligno

# Divida el conjunto de datos en entrenamiento y prueba.
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.4, random_state=1234)

# iniciar el KNN
knn = KNeighborsClassifier(n_neighbors=5)

# Entrenamiento del clasificador
knn.fit(X_train, Y_train)

# Predicciones
Y_pred = knn.predict(X_test)

# Exactitud
accuracy = metrics.accuracy_score(Y_test, Y_pred)

# Recuperacion
recall = metrics.recall_score(Y_test, Y_pred)

# Precisión
precision = metrics.precision_score(Y_test, Y_pred)

print("Accuracy | Exactitud:", accuracy)
print("Recall | recuperación:", recall)
print("Precisión:", precision)


Accuracy | Exactitud: 0.9254385964912281
Recall | recuperación: 0.8571428571428571
Precisión: 0.935064935064935


<Font color=red>[Respuesta]</Font>

Codigo:

# iniciar el KNN
knn = KNeighborsClassifier(n_neighbors=5)

# Entrenamiento del clasificador
knn.fit(X_train, Y_train)

# Predicciones
Y_pred = knn.predict(X_test)

# Exactitud
accuracy = metrics.accuracy_score(Y_test, Y_pred)

# Recuperacion
recall = metrics.recall_score(Y_test, Y_pred)

# Precisión
precision = metrics.precision_score(Y_test, Y_pred)

print("Accuracy:", accuracy)
print("Recall:", recall)
print("Precision:", precision)


# resultado

Accuracy: 0.9254385964912281

Recall: 0.8571428571428571

Precision: 0.935064935064935